ospfd/ospf_ext.c

```diff
@@ -264,18 +286,43 @@ static struct ext_itf *lookup_ext_by_ifp(struct interface *ifp)
 	return NULL;
 }
 
+/* Lookup prefix sid LSA by LSA ID from the extended prefix struct */
+static struct prefix_sid_lsa *
+lookup_psid_lsa_by_instance(const struct ext_itf *exti, const uint32_t instance)
+{
+	struct listnode *node;
+	struct prefix_sid_lsa *psid;
+
+	if (exti->type != OPAQUE_TYPE_EXTENDED_PREFIX_LSA)
+		return NULL;
+
+	for (ALL_LIST_ELEMENTS_RO(exti->lsa.prefix_sid_list, node, psid))
+		if (psid->instance == instance)
+			return psid;
+
+	return NULL;
```

Comment:
Hum! I'm bit lost here. The instance corresponds to the Opaque ID of the Opaque LSA. So, if an Extended Prefix Opaque LSA convey more than one prefix, all prefixes belongs to the same Opaque LSA, thus to the same instance. And there is one exti entry per interface. So, for a given loopback where you attach several IP address, thus several prefix SID, they are all convey in the same Opaque LSA with the same instance value. Note also that this instance value doesn't change when LSA is refresh.